### PR TITLE
Update docs in preparation for 1.0.0-alpha1 release

### PR DESCRIPTION
**Pull request by w01fe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,15 +4,20 @@ A Clojure(Script) library for declarative data description and validation.
 
 Leiningen dependency (Clojars): `[prismatic/schema "0.4.4"]`. [Latest codox API docs](http://prismatic.github.io/schema).
 
-**This is an alpha release. The API and organizational structure are
-subject to change. Comments and contributions are much appreciated.**
```
Comment (loganlinn) [inline README.md]: :confetti_ball:

File: README.md [hunk 2]
```diff
@@ -4,15 +4,20 @@ A Clojure(Script) library for declarative data description and validation.
...
 
-As of version 0.2.0, Schema also supports schema-driven data transformations, with *coercion* being the main application fleshed out thus far.  See [this post](http://blog.getprismatic.com/schema-0-2-0-back-with-clojurescript-data-coercion/) for a detailed overview, or check out the Coercion section below for an example.
+Schema is a rich language for describing data shapes, with a variety of features:
 
+ - Data validation, with descriptive error messages of failures (targeted at programmers)
+ - Annotation of function arguments and return values, with optional runtime validation
+ - Schema-driven data **coercion**, which can automatically, succinctly, and safely convert complex data types (see the Coercion section below)
+ - As of version 1.0.0, Schema also supports experimental `clojure.test.check` data **generation** from Schemas, as well as **completion** of partial datums, features we've found very useful when writing tests.
+ - Schema is also built into our [`plumbing`](https://github.com/Prismatic/plumbing) library, and our [`fnhouse`](https://github.com/Prismatic/fnhouse) library
+illustrates how we build APIs easily and safely with Schema.
```
Comment (loganlinn) [inline README.md]: :sheep: maybe "declaratively" instead of "safely"

Comment (loganlinn) [inline README.md]: :sheep: "into our plumbing and fnhouse libraries, which illustrates ..."

File: README.md [hunk 3]
```diff
@@ -295,9 +285,44 @@ Similarly, you can also write sequence schemas that expect particular values in
 (s/validate StringListOrKeywordMap [:foo])
 ;; RuntimeException:  Value does not match schema: [(not (instance? java.lang.String :foo))]
 
+;; if (shorthand for conditional)
+(def StringListOrKeywordMap (s/if map? {s/Keyword s/Keyword} [String]))
+
+;; cond-pre (experimental), also shorthand for conditional, allows you to skip the
+;; predicate when the options are superficially different
+(def StringListOrKeywordMap (s/cond-pre {s/Keyword s/Keyword} [String]))
+;; but don't do this (use `if` or `abstract-map-schema` here instead):
```
Comment (loganlinn) [inline README.md]: > but don't do this

might not clear why this is bad

File: README.md [hunk 4]
```diff
@@ -295,9 +285,44 @@ Similarly, you can also write sequence schemas that expect particular values in
...
+(def BadSchema (s/cond-pre {:a s/Keyword} {:b s/Keyword}))
+
+;; conditional can also be used to apply extra validation to a single type
+(def OddLong (s/conditional odd? long))
```
Comment (loganlinn) [inline README.md]: probably not the best example, as validating non-long value will throw from `odd?`

File: README.md [hunk 5]
```diff
@@ -333,11 +358,43 @@ There's nothing special about `json-coercion-matcher` though; it's just as easy
 
 For more details, see [this blog post](http://blog.getprismatic.com/schema-0-2-0-back-with-clojurescript-data-coercion/).
 
+## Generation and Completion
+
+As of version 1.0.0, Schema also provides two experimental forms of automatic test data generation from schemas.
```
Comment (loganlinn) [inline README.md]: :sheep: "also" isn't needed here

File: README.md [hunk 6]
```diff
@@ -105,12 +110,12 @@ What about when things go bad?  Schema's `s/check` and `s/validate` provide mean
 
 ```
 
-See the "More Examples" section below for more examples and explanation.
+See the "More Examples" section below for more examples and explanation, or the [custom schemas types](https://github.com/Prismatic/schema/wiki/Defining-New-Schema-Types-1.0) page for details on how Schema works under the hood.
```
Comment (loganlinn) [inline README.md]: "custom schemas types" -> "custom Schema types"

File: README.md [hunk 7]
```diff
@@ -333,11 +358,43 @@ There's nothing special about `json-coercion-matcher` though; it's just as easy
...
+;; => {:name "nL@", :barks? false, :type :dog}
+```
+
+The `schema.experimental.generators` namespace can compile Schemas into `clojure.test.check` generators.  All of the built-in
+schemas are supported out of the box, and it is easy to extend to add new types or customize generation on a per-type basis.
+See [`schema.experimental.generators-test`](https://github.com/Prismatic/schema/blob/master/test/clj/schema/experimental/generators_test.clj)
+for some more complex examples.
+
+Moreover, the `schema.experimental.complete` namespace can build on generation to allow "completion" of partial data.  Whereas generators and
+`clojure.test.check` are very useful tools for abstract property testing, `completers` are useful when we want to test the behavior of a
```
Comment (loganlinn) [inline README.md]: :sheep: `completers` or **completers** here?

File: README.md (hunk 1 above)
Comment (aria42) [inline README.md]: 🎓

Oh hai by the way guys. Great work!

**Comment by loganlinn** [comment]
LGMC!

File: README.md (hunk 3 above)
Comment (davegolland) [inline README.md]: :+1: 

maybe "`cond-pre` does a greedy match on the 'precondition' of the args, so if two schemas are structurally the same, then the checker will pick the first. For example, the following `BadSchema` will never validate `{:b :x}` because the first arg always wins"


(or maybe that's too wordy?)

File: README.md [hunk 8]
```diff
@@ -333,11 +358,43 @@ There's nothing special about `json-coercion-matcher` though; it's just as easy
...
+
+```clojure
+(require '[schema.experimental.complete :as c] '[schema.experimental.generators :as g])
+
+(g/sample 3 Animal)
+;; => ({:name "", :barks? false, :type :dog}
+;;     {:name "", :claws? false, :type :cat}
+;;     {:name "\"|", :claws? false, :type :cat})
+
+(g/generate Tree)
```
Comment (davegolland) [inline README.md]: awesome!

**Comment by davegolland** [comment]
:+1:

File: README.md (hunk 1 above)
Comment (w01fe) [inline README.md]: Hai @aria42 -- thanks!

File: README.md (hunk 4 above)
Comment (w01fe) [inline README.md]: see the `int` example below.
